gcloud-java-compute/src/test/java/com/google/gcloud/compute/SnapshotInfoTest.java

```diff
@@ -0,0 +1,135 @@
+  private static final List<LicenseId> LICENSES = ImmutableList.of(
+      LicenseId.of("project", "license1"), LicenseId.of("project", "license2"));
+  private static final SnapshotId SNAPSHOT_ID = SnapshotId.of("project", "snapshot");
+  private static final Status STATUS = Status.CREATING;
+  private static final Long DISK_SIZE_GB = 42L;
+  private static final String SOURCE_DISK_ID = "diskId";
+  private static final Long STORAGE_BYTES = 24L;
+  private static final StorageBytesStatus STORAGE_BYTES_STATUS = StorageBytesStatus.UP_TO_DATE;
+  private static final SnapshotInfo SNAPSHOT_INFO = SnapshotInfo.builder(SNAPSHOT_ID, SOURCE_DISK)
+      .id(ID)
+      .creationTimestamp(CREATION_TIMESTAMP)
+      .description(DESCRIPTION)
+      .status(STATUS)
+      .diskSizeGb(DISK_SIZE_GB)
+      .licenses(LICENSES)
+      .sourceDiskId(SOURCE_DISK_ID)
+      .storageBytes(STORAGE_BYTES)
+      .storageBytesStatus(STORAGE_BYTES_STATUS)
+      .build();
+
+  @Test
+  public void testToBuilder() {
+    compareSnapshotInfo(SNAPSHOT_INFO, SNAPSHOT_INFO.toBuilder().build());
+    SnapshotInfo snapshotInfo = SNAPSHOT_INFO.toBuilder().description("newDescription").build();
+    assertEquals("newDescription", snapshotInfo.description());
+    snapshotInfo = snapshotInfo.toBuilder().description("description").build();
+    compareSnapshotInfo(SNAPSHOT_INFO, snapshotInfo);
+  }
+
+  @Test
+  public void testToBuilderIncomplete() {
+    SnapshotInfo snapshotInfo = SnapshotInfo.of(SNAPSHOT_ID, SOURCE_DISK);
+    assertEquals(snapshotInfo, snapshotInfo.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(ID, SNAPSHOT_INFO.id());
+    assertEquals(SNAPSHOT_ID, SNAPSHOT_INFO.snapshotId());
+    assertEquals(CREATION_TIMESTAMP, SNAPSHOT_INFO.creationTimestamp());
+    assertEquals(DESCRIPTION, SNAPSHOT_INFO.description());
+    assertEquals(STATUS, SNAPSHOT_INFO.status());
+    assertEquals(DISK_SIZE_GB, SNAPSHOT_INFO.diskSizeGb());
+    assertEquals(LICENSES, SNAPSHOT_INFO.licenses());
+    assertEquals(SOURCE_DISK, SNAPSHOT_INFO.sourceDisk());
+    assertEquals(SOURCE_DISK_ID, SNAPSHOT_INFO.sourceDiskId());
+    assertEquals(STORAGE_BYTES, SNAPSHOT_INFO.storageBytes());
+    assertEquals(STORAGE_BYTES_STATUS, SNAPSHOT_INFO.storageBytesStatus());
+  }
+
+  @Test
+  public void testOf() {
+    SnapshotInfo snapshotInfo = SnapshotInfo.of(SNAPSHOT_ID, SOURCE_DISK);
+    assertNull(snapshotInfo.id());
+    assertEquals(SNAPSHOT_ID, snapshotInfo.snapshotId());
+    assertNull(snapshotInfo.creationTimestamp());
+    assertNull(snapshotInfo.description());
+    assertNull(snapshotInfo.status());
+    assertNull(snapshotInfo.diskSizeGb());
+    assertNull(snapshotInfo.licenses());
+    assertEquals(SOURCE_DISK, snapshotInfo.sourceDisk());
+    assertNull(snapshotInfo.sourceDiskId());
+    assertNull(snapshotInfo.storageBytes());
+    assertNull(snapshotInfo.storageBytesStatus());
+  }
+
+  @Test
+  public void testToAndFromPb() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/mziccard/gcloud-java/file/2293826574/issues/source?bid=3140968&fileBranchId=3191561#l103)